web/packages/teleterm/src/ui/ClusterConnect/ClusterLogin/useClusterLogin.ts

```diff
@@ -38,9 +40,12 @@ export default function useClusterLogin(props: Props) {
   const [passwordlessLoginState, setPasswordlessLoginState] =
     useState<PasswordlessLoginState>();
 
-  const [initAttempt, init] = useAsync(() =>
-    tshd.getAuthSettings({ clusterUri }).then(({ response }) => response)
-  );
+  const [initAttempt, init] = useAsync(() => {
+    return Promise.all([
+      tshd.getAuthSettings({ clusterUri }).then(({ response }) => response),
+      mainProcessClient.checkForAppUpdates(),
+    ]).then(r => r[0]);
```

Comment:
Thoughts on

```suggestion
    ]).then(([authSettings]) => authSettings);
```

(Not necessarily better because it's not the hardest thing to figure out what's going on, but it's a little bit clearer)